src/test/java/com/iopipe/__DoEventInfoPlugin__.java

```diff
@@ -0,0 +1,336 @@
+package com.iopipe;
+
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.CloudFrontEvent;
+import com.amazonaws.services.lambda.runtime.events.KinesisEvent;
+import com.amazonaws.services.lambda.runtime.events.KinesisFirehoseEvent;
+import com.amazonaws.services.lambda.runtime.events.S3Event;
+import com.amazonaws.services.lambda.runtime.events.ScheduledEvent;
+import com.amazonaws.services.lambda.runtime.events.SNSEvent;
+import com.amazonaws.services.s3.event.S3EventNotification;
+import com.amazonaws.services.simpleworkflow.flow.JsonDataConverter;
+import com.fasterxml.jackson.databind.cfg.MapperConfig;
+import com.fasterxml.jackson.databind.introspect.AnnotatedMethod;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.datatype.joda.JodaModule;
+import com.iopipe.http.RemoteRequest;
+import com.iopipe.http.RemoteResult;
+import com.iopipe.IOpipeMeasurement;
+import com.iopipe.plugin.eventinfo.EventInfoDecoder;
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.util.function.Supplier;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.Set;
+import javax.json.JsonArray;
+import javax.json.JsonNumber;
+import javax.json.JsonObject;
+import javax.json.JsonString;
+import javax.json.JsonValue;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+
+/**
+ * This tests that the event info plugin detects the input event sources for
+ * object correctly.
+ *
+ * @since 2018/04/16
+ */
+class __DoEventInfoPlugin__
+	extends Single
+{
+	/** Logging. */
+	private static final Logger _LOGGER =
+		LogManager.getLogger(__DoEventInfoPlugin__.class);
+	
+	/** Sent with no exception? */
+	protected final BooleanValue noerror =
```

Comment:
Trying to understand this initialization using `BooleanValue` -- help me out? These are initialized to false and then changed to `true` in the tests if they are successful, yes?

Reply:
Yes! This specific variable is set to `true` if an error was not thrown by the method that IOpipe was invoking.